conjure-lib/src/main/java/com/palantir/conjure/java/lib/SafeLong.java

```diff
@@ -25,8 +25,8 @@
 @Value.Immutable
 public abstract class SafeLong implements Comparable<SafeLong> {
 
-    private static final long MIN_SAFE_VALUE = -(1L << 53) + 1;
-    private static final long MAX_SAFE_VALUE = (1L << 53) - 1;
+    public static final long MIN_SAFE_VALUE = -(1L << 53) + 1;
```

Comment:
how about we expose a `Optional<SafeLong> safeValueOf(long value)` method instead?

Reply:
cc @carterkozak if he has thoughts

Reply:
We could do that, but I think we would still need to expose the max values, as is done with other numerical types (e.g. https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#MAX_VALUE).

If I _have_ to give a value, and safeValueOf gave me an empty optional, I need a way to determine what to report. I think it's relatively common to use the max value in that case, to represent the "you've gone past the limit" or "infinity" case.

Reply:
+1, I'm not sure what we'd expect folks to do in the `Optional.empty()` case without enough information to determine why the value is invalid.

Generally agree that it's helpful to expose the bounds in some way.

In terms of API design, what if we match `Long`, `Integer`, etc and provide two (new) public constants: `MAX_VALUE` and `MIN_VALUE`, but expose them as the `SafeLong` type?